service/config_provider.go

```diff
@@ -96,8 +103,25 @@ type ConfigMapConverterFunc func(*config.Map) error
 // defined by the given configFile and overwrites fields using properties.
 func NewDefaultConfigProvider(configFileName string, properties []string) ConfigProvider {
 	return NewConfigProvider(
-		[]configmapprovider.Provider{configmapprovider.NewFile(configFileName), configmapprovider.NewProperties(properties)},
-		[]ConfigMapConverterFunc{configprovider.NewExpandConverter()},
+		[]string{configFileName},
+		map[string]configmapprovider.Provider{
+			"file": configmapprovider.NewFile(),
+			"env":  configmapprovider.NewEnv(),
+		},
+		[]ConfigMapConverterFunc{
+			func(c *config.Map) error {
```

Comment:
It is not clear what this function does. It would be useful to extract it as a regular function, give it a name and document.

Reply:
Done.